flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdWindowProperties.scala

```diff
@@ -92,13 +93,57 @@ class FlinkRelMdWindowProperties private extends MetadataHandler[FlinkMetadata.W
       return null
     }
 
+    val windowSpec = childProps.getWindowSpec
+    // if the window a
+    if (!validateWindowProperties(windowSpec, mapInToOutPos)) {
```

Comment:
Nit: `validWindowProperties`. IMO, methods like `validateXXX` is usually with `void` as the return type. However, if the status is required, then the method name should be like `isValidXXX` or `validXXX`.